ambry-store/src/main/java/com/github/ambry/store/RemoteTokenTracker.java

```diff
@@ -145,14 +149,15 @@ public void close() {
    * Start the background persistor.
    */
   public void start(int persistIntervalInSeconds) {
-    if (persistIntervalInSeconds > 0) {
-      if (this.scheduler != null) {
-        persistorFuture = this.scheduler.scheduleAtFixedRate(persistor, new Random().nextInt(Time.SecsPerMin),
-            persistIntervalInSeconds, TimeUnit.SECONDS);
-        logger.info("flush peer's remote token every {} seconds", persistIntervalInSeconds);
-      } else {
-        logger.error("scheduler is null, couldn't persist peer's remote token.");
-      }
+    if (this.scheduler != null) {
+      persistorFuture =
+          this.scheduler.scheduleAtFixedRate(persistor, new Random().nextInt(Time.SecsPerMin), persistIntervalInSeconds,
+              TimeUnit.SECONDS);
+      logger.info("flush peer's remote token every {} seconds", persistIntervalInSeconds);
+    } else {
+      logger.error("scheduler is null, couldn't persist peer's remote token.");
+      // at least write once
+      persistor.run();
```

Comment:
this is not good enough, persist only once at the beginning won't be a real solution .

Reply:
It should happen either.
Because we check it in RemoteTokenTracker that schedule won't be null. 
But yes, it's confusing, I will remove the else condition and throw the exception in the constructor at the same  PR.

Reply:
Removed the else condition.